tests/unit_tests/conftest.py

```diff
@@ -938,7 +938,17 @@ def summary(self) -> pd.DataFrame:
         if self._summary is not None:
             return deepcopy(self._summary)
 
-        self._summary = self.get_file_contents("wandb-summary.json")
+        _summary = self.get_file_contents("wandb-summary.json")
+
+        # run summary may be updated multiple times,
+        # but we are only interested in the last one.
+        # we can have multiple runs saved to context,
+        # so we need to group by run id and take the
+        # last one for each run.
+        self._summary = (
```

Comment:
Didn’t we say the problem is with merge? It doesn’t seem like this is solving this, unless it is expected to have multiple lines per run I’d…